code/modules/organs/internal/liver/liver.dm

```diff
@@ -60,4 +60,26 @@
 	var/reagent_efficiency = 1
 	if(reagent_id in reagent_efficiencies)
 		reagent_efficiency = reagent_efficiencies[reagent_id]
-	H.reagents.remove_reagent(reagent_id, metabolism * efficiency * reagent_efficiency)
\ No newline at end of file
+	H.reagents.remove_reagent(reagent_id, metabolism * efficiency * reagent_efficiency)
+
+/datum/organ/internal/liver/diona
+	name = "filtration xylem"
+
+/datum/organ/internal/liver/diona/metabolize_reagent(var/reagent_id, var/metabolism)
+	..()
+	var/mob/living/carbon/human/H = owner
+	switch(reagent_id)
+		if(HYPERZINE || NUKA_COLA) // these are highly addictive and hallucinogenic drugs for diona - think sensory overload mixed with skooma
```

Comment:
@ComicIronic you're arguing that I copy paste this code instead?
In my opinion it's far far more reasonable that it be centralised in organ code rather than having copy pasted behaviour, as I would have to do for nuka cola and hyperzine. Honestly, race specific chemistry behaviour should have been based on the liver a long time ago. I'll accept your argument that it's not the normal behaviour and they should all be changed at the same time, maybe, but this is by far and away the better way to do this.